search.py

```diff
@@ -509,38 +509,40 @@ def and_search(states, problem, path):
     # body of and or search
     return or_search(problem.initial, problem, [])
 
+# Pre-defined actions for PeakFindingProblem
+primary_actions = { 'W':(-1,0), 'N':(0,1), 'E':(1,0), 'S':(0,-1) }
```

Comment:
- Better names would be something like `directions4` and `directions8`.
- There should be a space after a comma.
- do `directions8 = directions4 | {'NW': ... }

Reply:
The `|` operator cannot be used to append dicts. Should I do it in the following way:
directions8 = dict(directions4)
directions8.update({ 'NW': ... })